### PR TITLE
Upgrade slackclient

### PR DESCRIPTION
**Pull request by balajijegan**
- Upgrade slackclient
- The latest slack client had some modules refactored. Modify slackminion
to support the changes.
- Fix a unit test to use a DummySlackConnection.

### CODE REVIEW

File: tests/test_slack/test_user.py [hunk 1]
```diff
@@ -47,10 +47,10 @@ def test_repr(self):
...
 
     def test_get_user(self):
-        user = SlackUser.get_user(SlackClient('xxx'), test_user_name)
+        user = SlackUser.get_user(DummySlackConnection(), test_user_name)
         assert isinstance(user, SlackUser)
 
 
 def test_get_user_none():
-    user = SlackUser.get_user(SlackClient('xxx'), 'doesnotexist')
+    user = SlackUser.get_user(DummySlackConnection(), 'doesnotexist')
```
Comment (balajijegan) [inline tests/test_slack/test_user.py]: @arcticfoxnv  can you please review this? I was unsure why it was using SlackClient previously / how it worked.

File: slackminion/slack/user.py [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import logging
 
 
-class SlackUser(object):
+class  SlackUser(object):
```
Comment (arcticfoxnv) [inline slackminion/slack/user.py]: Remove the extra space

File: tests/test_slack/test_user.py (hunk 1 above)
Comment (arcticfoxnv) [reply tests/test_slack/test_user.py]: It's been a while since I looked at this.  Not sure why it was using SlackClient.

Comment (balajijegan) [reply tests/test_slack/test_user.py]: Addressed review comments. Thanks for the feedbac.

**Comment by balajijegan** [comment]
@arcticfoxnv  how does one get a new version released on pypi?

**Comment by balajijegan** [comment]
Fwiw pypi changed some repo format in April. Thanks for checking

**Comment by arcticfoxnv** [comment]
New version has been pushed.  It seems since that last time I pushed a version to pypi, the jenkins host I was using to handle it has been decommissioned and is long gone.

**Pull request merged**